src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Helper/ResultHelper.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerToys.Run.Plugin.WindowsSettings.Classes;
+using Microsoft.PowerToys.Run.Plugin.WindowsSettings.Properties;
+using Wox.Plugin;
+using Wox.Plugin.Logger;
+
+namespace Microsoft.PowerToys.Run.Plugin.WindowsRegistry.Helper
+{
+    /// <summary>
+    /// Helper class to easier work with results
+    /// </summary>
+    internal static class ResultHelper
+    {
+        /// <summary>
+        /// Return a list with <see cref="Result"/>s, based on the given list.
+        /// </summary>
+        /// <param name="list">The original result list to convert.</param>
+        /// <param name="iconPath">The path to the icon of each entry.</param>
+        /// <returns>A list with <see cref="Result"/>.</returns>
+        internal static List<Result> GetResultList(
+            in IEnumerable<IWindowsSetting> list,
+            string query,
+            in string iconPath)
+        {
+            var resultList = new List<Result>(list.Count());
+
+            foreach (var entry in list)
+            {
+                var type = Resources.ResourceManager.GetString($"{entry.Type}");
+
+                var result = new Result
+                {
+                    Action = (_) => DoOpenSettingsAction(entry),
+                    IcoPath = iconPath,
+                    SubTitle = $"{Resources.Area} \"{entry.Area}\" {Resources.SubtitlePreposition} {type}",
+                    Title = entry.Name,
+                    ContextData = entry,
+                };
+
+                AddOptionalToolTip(entry, result);
+
+                resultList.Add(result);
+            }
+
+            SetScores(resultList, query);
+
+            return resultList;
+        }
+
+        /// <summary>
+        /// Add a tool-tip to the given <see cref="Result"/>, based o the given <see cref="IWindowsSetting"/>.
+        /// </summary>
+        /// <param name="entry">The <see cref="IWindowsSetting"/> that contain informations for the tool-tip.</param>
+        /// <param name="result">The <see cref="Result"/> that need a tool-tip.</param>
+        private static void AddOptionalToolTip(IWindowsSetting entry, Result result)
+        {
+            var toolTipText = new StringBuilder();
+
+            var type = Resources.ResourceManager.GetString($"{entry.Type}");
```

Comment:
We should move the translation into `TranslateAllSettings()`. Then we can use user `entry.Type` directly.